itest/send_test.go

```diff
@@ -127,6 +127,98 @@ func testBasicSend(t *harnessTest) {
 	wg.Wait()
 }
 
+// testResumePendingPackageSend tests that we can properly resume a pending
+// package send after a restart.
+func testResumePendingPackageSend(t *harnessTest) {
+	ctxb := context.Background()
+
+	sendTapd := t.tapd
+
+	// Setup a receiver node.
+	recvLnd := t.lndHarness.Bob
+	recvTapd := setupTapdHarness(
+		t.t, t, recvLnd, t.universeServer,
+		func(params *tapdHarnessParams) {
+			// We expect the receiver node to exit with an error
+			// since it will fail to receive the asset at the first
+			// attempt. We will confirm that the receiver node does
+			// eventually receive the asset correctly via an RPC
+			// call.
+			params.expectErrExit = true
+		},
+	)
+
+	// Mint (and mine) an asset for sending.
+	rpcAssets := mintAssetsConfirmBatch(
+		t, sendTapd, []*mintrpc.MintAssetRequest{simpleAssets[0]},
+	)
+
+	genInfo := rpcAssets[0].AssetGenesis
+
+	// Synchronize the Universe state of the sending node, with the
+	// receiving node.
+	t.syncUniverseState(sendTapd, recvTapd, len(rpcAssets))
+
+	// The receiver node generates a new address.
+	recvAddr, err := recvTapd.NewAddr(
+		ctxb, &taprpc.NewAddrRequest{
+			AssetId: genInfo.AssetId,
+			Amt:     10,
+		},
+	)
+	require.NoError(t.t, err)
+	assertAddrCreated(t.t, recvTapd, rpcAssets[0], recvAddr)
+
+	// We will now start two asset send events in sequence. We will stop and
+	// restart the sending node during each send. During one sending event
+	// we will mine whilst the sending node is stopped. During the other
+	// sending event we will only mine once the sending node is restarted.
+	for i := range []int{0, 1} {
```

Comment:
This works as intended, even though `i` will contain the _index_, not the slice values. Could also just use the c-like syntax: `for i := 0; i < 2; i++ {`.

Reply:
Will fix in a future PR, thanks.

Reply:
Fixed in this PR: https://github.com/lightninglabs/taproot-assets/pull/371